### PR TITLE
Wait to avoid overlapping state saves

### PR DESCRIPTION
**Pull request by dqnykamp**

### CODE REVIEW

File: src/Core/Core.js [hunk 1]
```diff
@@ -11817,14 +11817,36 @@ export default class Core {
...
+    };
+
+    if (this.savingPageState) {
+      for (let i = 0; i < 100; i++) {
+        await pause100();
+
+        if (!this.savingPageState) {
+          break;
+        }
+      }
```
Comment (Lyanthropos) [inline src/Core/Core.js:11833]: Is there not an event message we can listen for?

File: src/Core/Core.js [hunk 2]
```diff
@@ -11841,7 +11863,10 @@ export default class Core {
       return;
     }
 
-    console.log("result from saving to database:", resp.data);
+    console.log(
+      `result from saving to database ${this.itemNumber}:`,
+      resp.data,
+    );
```
Comment (Lyanthropos) [inline src/Core/Core.js:11869]: I would leave this out or replace it why `console.debug`

**Review by Lyanthropos — APPROVED** [review]

File: src/Core/Core.js (hunk 2 above)
Comment (dqnykamp) [reply src/Core/Core.js]: removed

File: src/Core/Core.js (hunk 1 above)
Comment (dqnykamp) [reply src/Core/Core.js:11833]: Decided this is something to revisit later.

**Pull request merged**